src/adapter/http/tests/harness/client_side_harness.rs

```diff
@@ -48,6 +48,9 @@ impl ClientSideHarness {
         b.add::<auth::AlwaysHappyDatasetActionAuthorizer>()
             .bind::<dyn auth::DatasetActionAuthorizer, auth::AlwaysHappyDatasetActionAuthorizer>();
 
+        b.add::<SystemTimeSourceDefault>();
```

Comment:
@zaychenko-sergei  Note that I started introducing this interface to have more control over time in tests. Just letting you know it exists.

Reply:
It's a totally valid practice. Accessing "current time" directly is considered untestable. In languages like Python nobody cares, as they allow "freezing" time via dynamic language capabilities, but in compiled languages like Rust that would be the only valid way to deal with "current time".